examples/saas-clients/salesforce/working-with-salesforce-client/README.md

```diff
@@ -0,0 +1,256 @@
+- A Text Editor or an IDE 
+> **Tip**: For a better development experience, install one of the following Ballerina IDE plugins: 
+[VSCode](https://marketplace.visualstudio.com/items?itemName=ballerina.ballerina), 
+[IntelliJ IDEA](https://plugins.jetbrains.com/plugin/9520-ballerina)
+- Create a Salesforce account and create a connected app by visiting [Salesforce](https://www.salesforce.com) 
+and obtain the following parameters:
+    - Base URl (Endpoint)
+    - Client Id
+    - Client Secret
+    - Access Token
+    - Refresh Token
+    - Refresh URL
+
+- Visit 
+[here](https://help.salesforce.com/articleView?id=remoteaccess_authenticate_overview.htm) for more information on 
+obtaining OAuth2 credentials.
+
+Let's get started with a simple Ballerina program to integrate with Salesforce API.
+
+### Implementation
+The following diagram illustrates all the required functionality of the Salesforce Service that you are going to build.
+
+![Salesforce Guide Implementation](resources/salesforce_connector_guide_implementation.svg "Salesforce Guide Implementation")
+
+#### Creating the module structure
+Ballerina is a complete programming language that can have any custom project structure as you wish. Although the 
+language allows you to have any module structure, you can use the following simple module structure for this project.
+```
+working-with-salesforce-client
+    └── src
+         └── guide
+               └── tests
+               └── resources
+               └── ballerina.conf
+               └── working_with_salesforce_client.bal
+```
+- In the `working-with-salesforce-client/src/guide` directory there is a `ballerina.conf` file, which you should enter
+all the confugurations related to the guide. Add the following details you obtained above within the quotes.
+``` ballerina
+EP_URL = ""
+ACCESS_TOKEN = ""
+CLIENT_ID = ""
+CLIENT_SECRET = ""
+REFRESH_TOKEN = ""
+REFRESH_URL = ""
+```
+
+- Open a terminal, navigate to `working-with-salesforce-client` directory, and run the following Ballerina command to 
+run the `guide` module.
+```bash
+$ ballerina run --config src/guide/ballerina.conf guide
+```
+
+Then you will see salesforce service is up and running in the port 9090.
+
+#### Developing the service
+
+The following code is the completed sample which exposes the following services:
+- createAccount: Creates a new Account and related Contacts & Opportunities using the given JSON.
+- createContact: Creates a new Contact using given Contact details.
+- createOpportunity: Creates a new Opportunity using given Opportunity details.
+- executeQuery: Execute a given SOQL query and return results as a JSON.
+
+<!-- INCLUDE_CODE: guide/src/working_with_salesforce_client.bal -->
+
+### Deployment
+
+#### Deploying locally
+You can deploy the services that you developed above in your local environment. You can create the Ballerina executable 
+archives (.balx) first as follows.
+
+**Building**
```

Comment:
I feel this should be merged with the Testing section. Deployment should be primarily focused on how to deploy.